stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsSpec.kt

```diff
@@ -3,9 +3,13 @@ package io.getstream.chat.android.livedata.controller
 import io.getstream.chat.android.client.api.models.QuerySort
 import io.getstream.chat.android.client.models.Channel
 import io.getstream.chat.android.client.utils.FilterObject
+import java.util.Objects
 
 internal data class QueryChannelsSpec(
     val filter: FilterObject,
     val sort: QuerySort<Channel>,
     var cids: List<String> = emptyList()
-)
+) {
+    val id: String
+        get() = (Objects.hash(filter.toMap()) + Objects.hash(sort.toDto())).toString()
```

Comment:
why not `filter.toMap().hashCode()`?

Reply:
It's not new code, it was extracted from repository to model. Because it is used in business logic. Also, I think that queries with similar filter options and different sort ordering aren't same